### ISSUE TITLE
Does not seem to work with ES6 classes

### ISSUE THREAD

**Issue by jnystrom** [nkohari/forge#5]
The dependencies are not getting injected into the class `constructor` of the new ES6 class.  Type dependencies managed by Forge and ctor args using `with()` do not get injected.

**Comment by jnystrom** [comment]
Example: 

```
class SurpriseRepository {

    constructor(repository, ctorarg) {
        this.testService = repository;
        this.ctorarg = ctorarg;
    }

}

 forge.bind('repository').to.type(MongooseRepository);
 forge.bind('surpriseRepository').to.type(SurpriseRepository).with({ ctorarg: config.arg});
```

with this sort of registration...neither 'repository' nor 'ctorarg' will be available in the constructor of `SurpriseRepository`

**Comment by jnystrom** [comment]
@nkohari is this repo still active?

**Comment by nkohari** [comment]
@jnystrom It is, I just haven't had time to dig into the ES6 issues yet, since they don't affect what I'm doing. If you find a fix, a PR would be great... :)

**Comment by jnystrom** [comment]
@nkohari that is cool....I completely understand.  This repo is fantastic and, like you, if I can find time...I will try to help.  Thanks for all you do for us devs!

**Comment by teize001** [comment]
I don't know the coffee script. So I can not modify the source directly.
I am now using the typescript 1.8 and node v5.1.0.
So I found the solution but I'm not sure.

es6 uses a `class` and `constructor`.
So it seems that should not match the regular expression.

The following is a sample test code.

``` javascript
'use strict';

var Forge = require('forge-di');

// es6 class
class FooES6 {
    constructor (barES6) {
        this.bar = barES6;
    }

    test(item) {
        console.log('FooES6 test');
    }
}

class BarES6 {
    constructor () {  
        this.name = 'bar';
    }
}

// es5 class
function Foo(bar) {
    this.bar = bar;
}

function Bar() {
    this.name = 'bar';
}

var forge = new Forge();
forge.bind('foo').to.type(Foo);
forge.bind('bar').to.type(Bar);
forge.bind('fooES6').to.type(FooES6);
forge.bind('barES6').to.type(BarES6);

var obj = forge.get('foo');
console.dir(obj);
obj = forge.get('fooES6');
console.dir(obj);
```

modified code

``` javascript
// Inspector.js
...

    Inspector.prototype.getParameterNames = function(func) {
      var matches, regex;
      assert(func != null, 'The argument "func" must have a value');
      //regex = /function .*\(([^)]+)/g;      <== problem code ( not support es6 )
      regex = /(?:function|constructor)[ A-Za-z0-9]*\(([^)]+)/g;  // es5 or es6 support
      matches = regex.exec(func.toString());
      if ((matches == null) || matches[1].length === 0) {
        return [];
      }
      return matches[1].split(/[,\s]+/);
    };

...
```

**Comment by nkohari** [comment]
@teize001 Thanks so much! I recently needed to use Forge with TypeScript, so I was revisiting this problem. Your solution works great, and has been merged in to version 0.10.1.
